ml-agents/mlagents/trainers/settings.py

```diff
@@ -183,7 +183,7 @@ def to_settings(self) -> type:
 class RewardSignalSettings:
     gamma: float = 0.99
     strength: float = 1.0
-    normalize: bool = False
+    network_settings: NetworkSettings = attr.ib(factory=NetworkSettings)
```

Comment:
I would make this one optional and if it is None, then use the Policy's network settings rather than our own defaults. How does that sound ?

Reply:
I think I'd prefer to use our defaults since it's possible the policy has significantly more capacity than is needed i.e. the Crawler policy of 3/512 vs what we use for the discriminator 2/128. That being said, I also realize this enables users to specify `memory` which we probably want to explicitly prevent in the reward providers. cc @ervteng

Reply:
Not opposed to either route, they have their own pros/cons. Either way as long as it's documented it should be fine. 
Is getting the Policy settings super ugly?

Reply:
Im not sure how future proof it is for multi-agent scenarios. We could have different policies to select from. Additionally, we currently create reward signals in the `optimizer/torch_optimizer.py` and in the future i think it will be necessary to remove the policy from the optimizer (also for multiagent) in which case this would need to be addressed by either keeping the policy around/moving the creation of the reward provider. My vote is for default network settings